pkg/dt/fdt.go

```diff
@@ -71,6 +80,36 @@ type ReserveEntry struct {
 	Size    uint64
 }
 
+// LoadFDTFromFile loads a flateen device tree blob.
+//
+// It first tries to load it from given file path, then
+// from /sys/firmware/fdt. If still failed, it resorts
+// to read /proc/device-tree, and flatten it.
+func LoadFDTFromFile(dtbPath string) (*FDT, error) {
```

Comment:
It seems like there is more than one function here. If you were to just have this load from one file, only, then you could define another function that uses a variadic.

Further, having this function implicitly load from one place if it can't open another place is going to confuse people, I suspect.

Reply:
For example, if this took a variadic: 
func LoadFDTFromFile(arg ...string)

then people could get what they want this way:
LoadFDT("xyz.fdt", someotherpath, whatever)